docs/source/install.rst

```diff
@@ -200,83 +248,74 @@ If you want to use cuDNN or NCCL installed in other directory, please use ``CFLA
 .. note::
 
    Use full paths for the environment variables.
-   ``distutils`` that is used in the setup script does not parse the home directory mark ``~``.
-
+   ``distutils`` that is used in the setup script does not expand the home directory mark ``~``.
 
-Install CuPy for developers
-~~~~~~~~~~~~~~~~~~~~~~~~~~~
-
-CuPy uses Cython (>=0.26.1).
-Developers need to use Cython to regenerate C++ sources from ``pyx`` files.
-We recommend to use ``pip`` with ``-e`` option for editable mode::
-
-  $ pip install -U cython
-  $ cd /path/to/cupy/source
-  $ pip install -e .
+.. _install_cuda:
 
-Users need not to install Cython as a distribution package of CuPy only contains generated sources.
+Working with Custom CUDA Installation
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
+If you have installed CUDA on the non-default directory or have multiple CUDA versions installed, you may need to manually specify the CUDA installation directory to be used by CuPy.
 
-Uninstall CuPy
---------------
+CuPy uses the first CUDA installation directory found by the following order.
 
-Use pip to uninstall CuPy::
+#. ``CUDA_PATH`` environment variable.
+#. The parent directory of ``nvcc`` command. CuPy looks for ``nvcc`` command in each directory set in ``PATH`` environment variable.
+#. ``/usr/local/cuda``
 
-  $ pip uninstall cupy
-
-.. note::
+For example, you can tell CuPy to use non-default CUDA directory by ``CUDA_PATH`` environment variable::
 
-   When you upgrade Chainer, ``pip`` sometimes install the new version without removing the old one in ``site-packages``.
-   In this case, ``pip uninstall`` only removes the latest one.
-   To ensure that Chainer is completely removed, run the above command repeatedly until ``pip`` returns an error.
+  $ CUDA_PATH=/opt/nvidia/cuda pip install cupy
 
 .. note::
 
-   If you installed CuPy using wheel, use ``pip uninstall cupy-cudaXX`` (where XX is a CUDA version number) instead.
+   CUDA installation discovery is also performed at runtime using the rule above.
+   Depending on your system configuration, you may also need to set ``LD_LIBRARY_PATH`` environment variable to ``$CUDA_PATH/lib64`` at runtime.
 
-Upgrade CuPy
-------------
+Using custom ``nvcc`` command during installation
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-Just use ``pip`` with ``-U`` option::
+If you want to use a custom ``nvcc`` compiler (for example, to use ``ccache`` ), please set ``NVCC`` environment variables before installing CuPy::
 
-  $ pip install -U cupy
+  export NVCC='ccache nvcc'
 
+.. note::
 
-Reinstall CuPy
---------------
+   Setting ``NVCC`` environment variable does not affect at runtime, as CuPy does not use ``nvcc`` command at runtime.
```

Comment:
What do you think about this alternative? `During runtime, you don't need to set this environment variable since CuPy doesn't use the nvcc command.`